### PR TITLE
[Project] Change config defines to be prefixed

### LINKED ISSUE #4143: Build configuration names are UPPER_CASED and set as preprocessor macros, which can cause build errors
CocoaPods takes build configurations like **Debug** and **Release** and auto-magically snakecases and uppercases their names to create preprocessor macros like **DEBUG** and **RELEASE**. This is problematic if you include a pod that is not expecting one of these macros to exist, and can result in an error such as the following (sample project [here](http://cl.ly/3U0m0P3b2G2x/CocoaPodsBuildConfigurationPreprocessorBug.zip)):

screen shot 2015-09-08 at 8 36 16 pm

According to @mrackwitz this can be worked around by clobbering these macros in post-install hook, though I have not yet figured out how to actually do this.

### PR DESCRIPTION
**Pull request by segiddins**
Closes #4143.
Closes #4148.

\c @mrackwitz

### CODE REVIEW

File: lib/cocoapods/project.rb [hunk 1]
```diff
@@ -240,18 +240,29 @@ def add_podfile(podfile_path)
...
       build_configuration
     end
 
+    # @param  [String] name
+    #         The name of the build configuration.
+    #
+    # @return [String] The preprocessor definition to set for the configuration.
+    #
+    def define_for_build_configuration(name)
+      "POD_CONFIGURATION_#{name}".gsub(/[^a-zA-Z0-9_]/, '_').upcase
```
Comment (mrackwitz) [inline lib/cocoapods/project.rb]: Let's prefix that with `.underscore`, so that camelcase configuration names will yield to a nicer output.

Comment (segiddins) [inline lib/cocoapods/project.rb]: done

**Comment by segiddins** [comment]
This is a breaking change, please review?

File: lib/cocoapods/project.rb [hunk 2]
```diff
@@ -240,18 +241,29 @@ def add_podfile(podfile_path)
...
       settings['GCC_PREPROCESSOR_DEFINITIONS'] = definitions
       build_configuration
     end
 
+    # @param  [String] name
+    #         The name of the build configuration.
+    #
+    # @return [String] The preprocessor definition to set for the configuration.
+    #
+    def define_for_build_configuration(name)
```
Comment (mrackwitz) [inline lib/cocoapods/project.rb]: As this method doesn't have any side-effect, I don't think it should be named by the leading verb convention. I'd prefer `definition_for_…` here.

**Comment by mrackwitz** [comment]
:+1: LGTM beside my last comment

**Comment by neonichu** [comment]
LGTM

**Pull request merged**